shared/actions/users.tsx

```diff
@@ -44,9 +45,37 @@ const getBio = async (state: TypedState, action: UsersGen.GetBioPayload) => {
   }
 }
 
+const setUserBlocks = async (_: TypedState, action: UsersGen.SetUserBlocksPayload) => {
+  const {blocks} = action.payload
+  await RPCTypes.userSetUserBlocksRpcPromise({blocks}, Constants.setUserBlocksWaitingKey)
+}
+
+const getBlockState = async (_: TypedState, action: UsersGen.GetBlockStatePayload) => {
+  const {usernames} = action.payload
+
+  const userBlocks = await RPCTypes.userGetUserBlocksRpcPromise({usernames})
+  if (userBlocks) {
+    return userBlocks.map(block =>
+      UsersGen.createUpdateBlockState({
```

Comment:
Could this take an array payload? It'd be better to only update the store one time with the result of this.